ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -73,26 +74,50 @@ export default Route.extend(UnloadModelRoute, {
       secret = secret.replace('cert/', '');
     }
     return hash({
-      secret: this.store.queryRecord(modelType, { id: secret, backend }).then(resp => {
-        if (modelType === 'secret-v2') {
-          let backendModel = this.modelFor('vault.cluster.secrets.backend', backend);
-          let targetVersion = parseInt(params.version || resp.currentVersion, 10);
-          let version = resp.versions.findBy('version', targetVersion);
-          // 404 if there's no version
-          if (!version) {
-            let error = new DS.AdapterError();
-            set(error, 'httpStatus', 404);
-            throw error;
-          }
-          resp.set('engine', backendModel);
+      secret: this.store
+        .queryRecord(modelType, { id: secret, backend })
+        .then(secretModel => {
+          if (modelType === 'secret-v2') {
+            let targetVersion = parseInt(params.version || secretModel.currentVersion, 10);
+            let version = secretModel.versions.findBy('version', targetVersion);
+            // 404 if there's no version
+            if (!version) {
+              let error = new DS.AdapterError();
+              set(error, 'httpStatus', 404);
+              throw error;
+            }
+            secretModel.set('engine', backendModel);
 
-          return version.reload().then(() => {
-            resp.set('selectedVersion', version);
-            return resp;
-          });
-        }
-        return resp;
-      }),
+            return version.reload().then(() => {
+              secretModel.set('selectedVersion', version);
+              return secretModel;
+            });
+          }
+          return secretModel;
+        })
+        .catch(err => {
+          //don't have access to the metadata, so we'll make
+          //a dummy metadata model and try to load the version
```

Comment:
can we use a word other than dummy?

Reply:
Sure - "stub"/"isStub" maybe? 🤔 Think of any other better ones?